app/controllers/auth/registrations_controller.rb

```diff
@@ -1,13 +1,22 @@
 # frozen_string_literal: true
 # Auth::RegistrationsController manages user signup/removal.
 class Auth::RegistrationsController < Devise::RegistrationsController
+  before_action :restrict_admin
+
   layout "authentication", except: :edit
 
   skip_before_action :redirect_to_setup
 
   # Re-implemented so the template has some auxiliary instance variables.
   def new
+    binding.pry
     @have_users = User.any?
     super
   end
+
+  def restrict_admin
+    return if User.count == 0
+    flash[:alert] = "Admin user already exists."
+    redirect_to root_path
+  end
```

Comment:
We could use a regular flow here since complexity is very low, why not:

```
if User.count > 0
  flash[:alert] = "Admin user already exists."
  redirect_to root_path
end
```